src/Lib.hs

```diff
@@ -95,27 +97,39 @@ fetchAgents = do
 
 processTicket :: TicketId -> App (Maybe ZendeskResponse)
 processTicket tId = do
+    printText <- asksIOLayer iolPrintText
+    -- Printing id before inspecting the ticket so that when the process stops by the
+    -- corrupted log file, we know which id to blacklist.
+    printText $ "Analyzing ticket id: " <> show (getTicketId tId)
 
     -- We first fetch the function from the configuration
     getTicketInfo       <- asksZendeskLayer zlGetTicketInfo
-    printText           <- asksIOLayer iolPrintText
-
-    printText $ "Processing a ticket: " <> show (getTicketId tId)
-
     mTicketInfo         <- getTicketInfo tId
+    
     getTicketComments   <- asksZendeskLayer zlGetTicketComments
     comments            <- getTicketComments tId
     let attachments     = getAttachmentsFromComment comments
     let ticketInfo      = fromMaybe (error "No ticket info") mTicketInfo
     zendeskResponse     <- getZendeskResponses comments attachments ticketInfo
-    postTicketComment   <- asksZendeskLayer zlPostTicketComment
-    whenJust zendeskResponse postTicketComment
 
-    printText "Process finished, please see the following url"
-    printText $ "https://iohk.zendesk.com/agent/tickets/" <> show (getTicketId tId)
+    postTicketComment   <- asksZendeskLayer zlPostTicketComment
+    
+    whenJust zendeskResponse $ \response -> do
+        let formattedTags = formatZendeskResponseTags response
+        printText formattedTags
+        appendF <- asksIOLayer iolAppendFile
+        appendF "logs/analysis-result.log" (show (getTicketId tId) <> " " <> formattedTags <> "\n")
```

Comment:
I don't like this `append` since it could introduce issues due to file changes, it could be easier to add it all in a String/Text/ByteString and add it into the file at the end of the function. These mutations over time always end up in some weird situations...

There are other issues connected with that, like losing log info due to program crashing or large overhead in memory (really negligible), but I guess it's more obvious/transparent to the user if there is an issue in the application, there should be one in the logs as well.

Other than that, kudos for updating our IO layer. A bonus question would be - how would you test `append` or the function? Does `append` reflect the function behaviour? :thinking: 
Seems like there is something there :wink: